### PR TITLE
Improve clarity on application configuration

### PR DESCRIPTION
**Pull request by aduth**
**Why**:

Improve clarity on...

- Where do the default values come from
- How are they overridden
- The ability to override values in local development

### CODE REVIEW

File: _articles/appdev-secrets-configuration.md [hunk 1]
```diff
@@ -8,16 +8,16 @@ subcategory: Development
...
+environment by merging default values with an environment-specific YAML file.
 
+* In local development, this file lives at `config/application.yml` and is created during setup.
+* In deployed environments, this file is downloaded from S3 when activating or deploying an instance
+  (see [`deploy/activate`][deploy-activate] and the [`activate.rb`][download-from-s3]).
 
-We store these `application.yml` files in S3, and each environment (`int`, `dev`)
-has its own separate file, so secrets and configs need to be explicitly copied
-across environments, they are not implicitly shared.
+**Changing configuration for a deployed application requires a recycle**, since this merge step only
```
Comment (zachmargolis) [inline _articles/appdev-secrets-configuration.md:19]: should we link the deploy/recycle article here?

**Review by zachmargolis — APPROVED** [review]
LGTM

File: _articles/appdev-secrets-configuration.md (hunk 1 above)
Comment (aduth) [reply _articles/appdev-secrets-configuration.md:19]: Makes sense to me! There's even a configuration-specific recycle section we could deep-link to:

https://handbook.login.gov/articles/appdev-deploy.html#config-recycle

File: _articles/appdev-secrets-configuration.md [hunk 2]
```diff
@@ -8,27 +8,28 @@ subcategory: Development
...
-[deploy-activate]: https://github.com/18F/identity-idp/blob/main/deploy/activate
-[download-from-s3]: https://github.com/18F/identity-idp/blob/a95fd33d24c6761818993cfbc334a28986783034/lib/deploy/activate.rb#L93-L97
-
 At the end of the day, since these are just files in S3, you can use whatever workflow
 you want to download, edit, and write them. Make sure you clean up files on your local
 machine when done.
 
+[deploy-activate]: https://github.com/18F/identity-idp/blob/main/deploy/activate
+[download-from-s3]: https://github.com/18F/identity-idp/blob/a95fd33d24c6761818993cfbc334a28986783034/lib/deploy/activate.rb#L93-L97
+[recycle-config]: ./appdev-deploy.html#config-recycle
```
Comment (zachmargolis) [inline _articles/appdev-secrets-configuration.md:31]:
```suggestion
[recycle-config]: {% link _articles/appdev-deploy.md %}#config-recycle
```

Comment (aduth) [reply _articles/appdev-secrets-configuration.md]: Thanks! Forgot about that helper.

**Pull request merged**